vizro-core/examples/_chart-gallery/utils/custom_extensions.py

```diff
@@ -0,0 +1,116 @@
+"""Contains custom components and charts used inside the dashboard."""
+
+from typing import List, Literal
+
+import dash_bootstrap_components as dbc
+import pandas as pd
+import plotly.graph_objects as go
+import vizro.models as vm
+from dash import dcc, html
+from vizro.models.types import capture
+
+try:
+    from pydantic.v1 import Field
+except ImportError:  # pragma: no cov
+    from pydantic import Field
+
+
+class CodeClipboard(vm.VizroBaseModel):
```

Comment:
@antonymilne - What do you think? Initially, we discussed having it only in the example code. I could also add it to the custom components docs page if that would be useful. Do we want to make this a proper custom component within our source code at some point, maybe later?

Reply:
As discussed, maybe yes in our codebase at some point but not for a while I think 🙂 I did already make some changes to this to make it a bit more generic/less written for this example dashboard though. And some version of this would be a great docs example, I really like it 👍